pkg/claimsstore/jwtclaimpersister.go

```diff
@@ -0,0 +1,105 @@
+package claimsstore
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/jinzhu/gorm"
+	"github.com/joincivil/id-hub/pkg/didjwt"
+	"github.com/multiformats/go-multihash"
+	didlib "github.com/ockam-network/did"
+	"github.com/pkg/errors"
+)
+
+// JWTClaimPostgres is a model for storing jwt type vcs
+type JWTClaimPostgres struct {
+	JWT    string `gorm:"not null"`
+	Issuer string `gorm:"not null;index:jwtissuer"`
+	Sender string `gorm:"not null;index:jwtsender"`
+	Hash   string `gorm:"primary_key"`
+}
+
+// TableName sets the name of the table in the db
+func (JWTClaimPostgres) TableName() string {
+	return "jwt_claims"
+}
+
+func hashJWT(token string) (string, error) {
+	hash := crypto.Keccak256([]byte(token))
+	mHash, err := multihash.EncodeName(hash, "keccak-256")
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(mHash), nil
+}
+
+// JWTClaimPGPersister is a postgres persister for JWT claims
+type JWTClaimPGPersister struct {
+	db            *gorm.DB
+	didJWTService *didjwt.Service
+}
+
+// NewJWTClaimPGPersister returns a new JWTClaimPGPersister
+func NewJWTClaimPGPersister(db *gorm.DB, didJWTService *didjwt.Service) *JWTClaimPGPersister {
+	return &JWTClaimPGPersister{
+		db:            db,
+		didJWTService: didJWTService,
+	}
+}
+
+// AddJWT adds a new jwt claim to the db
+func (p *JWTClaimPGPersister) AddJWT(tokenString string, senderDID *didlib.DID) (string, error) {
+	token, err := p.didJWTService.ParseJWT(tokenString)
+	if err != nil {
+		return "", err
```

Comment:
Do we need to wrap some of these errors?

Reply:
will do